### PR TITLE
Code quality

### LINKED ISSUE #352: ThreadPoolPushButton test fails
I get this traceback:

```
ERROR:silx.gui.widgets.ThreadPoolPushButton:Error while executing callable silx.gui.widgets.test.test_threadpoolpushbutton._computeFail.
Traceback (most recent call last):
  File "/Users/marius/Software/silx/build/lib.macosx-10.11-x86_64-3.5/silx/gui/widgets/ThreadPoolPushButton.py", line 56, in run
    result = self.__callable(*self.__args, **self.__kwargs)
  File "/Users/marius/Software/silx/build/lib.macosx-10.11-x86_64-3.5/silx/gui/widgets/test/test_threadpoolpushbutton.py", line 54, in _computeFail
    raise Exception("exception")
Exception: exception
```

Is this error supposed to be logged at the end of the test? I get this:

```
----------------------------------------------------------------------
Ran 691 tests in 62.155s

OK (skipped=1)
```

### PR DESCRIPTION
**Pull request by vallsv**
- Filter and test logging message (close #352)
- Ignore unreleased error widgets if the test fail
- Fix bunch of warnings (unused var, lib, space, trailing space...)
- Refactoring main function from ImageView as an example

### CODE REVIEW

File: examples/fftPlotAction.py [hunk 1]
```diff
@@ -105,8 +105,8 @@ def fftAllCurves(self, checked=False):
 
         self.plot.clearCurves()
 
-        for i, curve in enumerate(allCurves):
-            x, y, legend, info, params = curve
+        for _i, curve in enumerate(allCurves):
```
Comment (t20100) [inline examples/fftPlotAction.py]: `_i` seems unused, why not just` for curve in allCurves:`

**Comment by vallsv** [comment]
Fixed

File: examples/fftPlotAction.py (hunk 1 above)
Comment (PiRK) [inline examples/fftPlotAction.py]: Probably a remnant from a past version. I agree that `i` and `enumerate` are no longer necessary.

**Comment by vallsv** [comment]
After talking to Pierre, i remove changes on indentation (our editors never agreed with each others).

**Pull request merged**